servers/rendering/renderer_rd/shaders/forward_clustered/scene_forward_clustered.glsl

```diff
@@ -311,14 +314,13 @@ void vertex_shader(vec3 vertex_input,
 #endif
 		//transpose
 		matrix = transpose(matrix);
-#if !defined(USE_DOUBLE_PRECISION) || defined(SKIP_TRANSFORM_USED) || defined(VERTEX_WORLD_COORDS_USED) || defined(MODEL_MATRIX_USED)
```

Comment:
I'm not sure about this. At a glance, this appears to change the behaviour of shaders that dont skip vertex transform, don't use world coords, and don't read from the model matrix. Have you tested all the permutations and confirmed this doesn't break non-doubles builds?

Reply:
I remember puzzling over this particularly configuration set and deciding they weren't necessary. I can't remember why now. I'll have to verify this with custom shaders and make sure there's no regression or bring back the conditionals. Is `MODEL_MATRIX_USED` set automatically when a custom shader accesses the model matrix?

Reply:
Specifically, my logic was:
> The matrix can be baked into the modelview matrix in both double and single precision cases. This removes the need for some preprocessor macros since now the model_matrix will be used in both configurations. For double precision, the translation component of matrix is removed to be used in the high precision calculations after the material's vertex shader.
Clear as mud. I'm assuming that a billboarding shader will always set `MODEL_MATRIX_USED` as true since it writes to the model matrix?

Reply:
Hmmmm, it's curious that [this workaround](https://github.com/godotengine/godot/issues/73383#issuecomment-1890991699) also involves stripping these conditionals. Is there some documentation on what each of these definitions `MODEL_MATRIX_USED`, `SKIP_TRANSFORM_USED`, and `VERTEX_WORLD_COORDS_USED` mean or maybe is you know where they are set? I'd like to dig a bit deeper as part of this issue and #73383.

Reply:
`MODEL_MATRIX_USED` is defined when a user uses `MODEL_MATRIX` in their shader.

`SKIP_TRANSFORM_USED` is defined when a user uses the `skip_vertex_transform` render_mode

`VERTEX_WORLD_COORDS_USED` is defined when a user uses the `world_vertex_coords` render_mode